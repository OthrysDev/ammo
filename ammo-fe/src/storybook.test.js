import React from 'react';
import initStoryshots, {
    multiSnapshotWithOptions,
} from '@storybook/addon-storyshots';
import { StylesProvider } from '@material-ui/styles';
import { addDecorator } from '@storybook/react';
import ReactDOM from 'react-dom';

//* Might be deprecated at some point when official packaged is released : https://github.com/enzymejs/enzyme/issues/2429
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

//* We need to use enzyme for storybook to handle properly React Portals feature
import { mount, configure } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
configure({ adapter: new Adapter() });

// To maintain consistency in material generated class names : https://github.com/mui-org/material-ui/issues/9492
const generateClassName = (rule, styleSheet) =>
    `${styleSheet.options.classNamePrefix}-${rule.key}`;

addDecorator((storyFn) => (
    <StylesProvider generateClassName={generateClassName}>
        {storyFn()}
    </StylesProvider>
));

//* Need serializer AND the renderer, see documentation at 'renderer' : https://www.npmjs.com/package/@storybook/addon-storyshots
initStoryshots({
    snapshotSerializers: [createSerializer()],
    test: multiSnapshotWithOptions({
        renderer: mount,
    }),
});
