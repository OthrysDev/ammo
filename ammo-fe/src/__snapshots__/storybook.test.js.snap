// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots MainPannel Default 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-root-3"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-6 makeStyles-root-5"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/0
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
</div>
`;

exports[`Storyshots MainPannel Empty 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 makeStyles-root-1"
/>
`;

exports[`Storyshots MainPannel Rich 1`] = `
<div
  className="MuiBox-root MuiBox-root-8 makeStyles-root-7"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-10 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/0
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-11 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/1
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-12 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/2
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-13 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/3
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-14 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/4
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-15 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/5
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-16 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/6
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-17 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/7
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-18 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/8
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <div
        className="MuiBox-root MuiBox-root-19 makeStyles-root-9"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          ::
          0
           
          GET
           |
           
          http://localhost/foo/9
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      script
    </div>
  </div>
</div>
`;
